rotkehlchen/chain/evm/decoding/xdai_bridge/decoder.py

```diff
@@ -0,0 +1,125 @@
+            bridge_address: ChecksumEvmAddress,
+            bridged_asset: 'Asset',
+            source_chain: ChainID,
+            target_chain: ChainID,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridge_address = bridge_address
+        self.bridged_asset = bridged_asset
+        self.bridged_crypto_asset = bridged_asset.resolve_to_crypto_asset()
+        self.bridged_asset_with_symbol = bridged_asset.resolve_to_asset_with_symbol()
+        self.deposit_topic = deposit_topic
+        self.withdrawal_topic = withdrawal_topic
+        self.source_chain = source_chain
+        self.target_chain = target_chain
+
+    def _decode_bridged_asset(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a bridging event for the `bridged_asset`, either a deposit or a withdrawal."""
+        if context.tx_log.topics[0] == self.deposit_topic:
+            from_address = context.transaction.from_address
+            to_address = self.bridge_address
+        elif context.tx_log.topics[0] == self.withdrawal_topic:
+            from_address = self.bridge_address
+            to_address = hex_or_bytes_to_address(context.tx_log.data[0:32])
+        else:
+            return DEFAULT_DECODING_OUTPUT
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.bridged_crypto_asset,
+        )
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_events=(self.deposit_topic,),
+            source_chain=self.source_chain,
+            target_chain=self.target_chain,
+            from_address=from_address,
+            to_address=to_address,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == expected_event_type and
+                event.location_label == expected_location_label and
+                event.address == self.bridge_address and
+                event.asset == self.bridged_asset and
+                (amount is None or event.balance.amount == amount)
+            ):
+                bridge_match_transfer(
+                    event=event,
+                    from_address=context.transaction.from_address,
+                    to_address=hex_or_bytes_to_address(context.tx_log.data[0:32]),
+                    from_chain=from_chain,
+                    to_chain=to_chain,
+                    amount=event.balance.amount,
+                    asset=self.bridged_asset_with_symbol,
+                    expected_event_type=expected_event_type,
+                    new_event_type=new_event_type,
+                    counterparty=XDAI_BRIDGE_CPT_DETAILS,
+                )
+                return DEFAULT_DECODING_OUTPUT
+        log.error(
```

Comment:
Why did you not do it as I said. This way you now have two returns. Please do what I said with the break and the else in the for.

Reply:
the break and else in the for doesn't work, that's why I did this.

Reply:
if I do an else, it returns the log for every event that's not the transfer event